internal/dependency/health_service.go

```diff
@@ -28,6 +29,11 @@ var (
 
 	// HealthServiceQueryRe is the regular expression to use.
 	HealthServiceQueryRe = regexp.MustCompile(`\A` + tagRe + serviceNameRe + dcRe + nearRe + filterRe + `\z`)
+
+	// queryParamOptRe is the regular expression to distinguish between query
+	// params and filters. Query parameters only have one "=" where as filters
+	// can have "==" or "!=" operators.
+	queryParamOptRe = regexp.MustCompile(`\w*[^!]={1}\w*`)
```

Comment:
I'm not great with regex and played around with this. When I tried this out, `queryParamOptRe.MatchString(opt) ` returned true for "==" use case e.g. `Checks.Status == passing`. It seems like this should return false, but let me know if I misunderstand!

Reply:
Looks like I'm not great either 😝! This regex doesn't do exactly what I intended, now i'm wondering why the tests pass.

```
[\w\d\s]=[\w\d\s]
```

Reply:
`[\w\d\s]=[\w\d\s]` works! It's also very easy to read, I like it. Also, thanks for the helpful comment on the regex. I see a "===" filters test in TestNewHealthServiceQueryV1. Maybe that is supposed to be the "==" test?

Reply:
Ah the `===` was testing for bad go-bexpr grammer and I didn't even test the regex. Thank you for pointing it out! The tests added in the new commit https://github.com/hashicorp/hcat/pull/35/commits/2338397c65c70333ead0f4e17481336dfbb978e6 fails with the current regex and now passes with the new one.

Reply:
👍 test looks great!